test/neural_networks/test_sampler_qnn.py

```diff
@@ -0,0 +1,279 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test Sampler QNN with Terra primitives."""
+
+from test import QiskitMachineLearningTestCase
+
+import itertools
+import unittest
+import numpy as np
+
+from ddt import ddt, idata, unpack
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.primitives import Sampler
+from qiskit.algorithms.gradients import ParamShiftSamplerGradient
+from qiskit.circuit.library import RealAmplitudes, ZZFeatureMap
+from qiskit.utils import QuantumInstance, algorithm_globals
+
+from qiskit_machine_learning.neural_networks import CircuitQNN
+from qiskit_machine_learning.neural_networks.sampler_qnn import SamplerQNN
+import qiskit_machine_learning.optionals as _optionals
+
+algorithm_globals.random_seed = 42
+
+DEFAULT = "default"
+SHOTS = "shots"
+SPARSE = [True, False]
+SAMPLERS = [DEFAULT, SHOTS]
+INTERPRET_TYPES = [2]
+BATCH_SIZES = [1, 2]
+
+
+@ddt
+class TestSamplerQNN(QiskitMachineLearningTestCase):
+    """Sampler QNN Tests."""
+
+    def setUp(self):
+        super().setUp()
+        algorithm_globals.random_seed = 12345
+
+        # # define test circuit
+        # num_qubits = 3
+        # self.qc = RealAmplitudes(num_qubits, entanglement="linear", reps=1)
+
+        # define feature map and ansatz
+        num_qubits = 2
+        feature_map = ZZFeatureMap(num_qubits, reps=1)
+        var_form = RealAmplitudes(num_qubits, reps=1)
+
+        # construct circuit
+        self.qc = QuantumCircuit(num_qubits)
+        self.qc.append(feature_map, range(2))
+        self.qc.append(var_form, range(2))
+
+        # store params
+        self.input_params = list(feature_map.parameters)
+        self.weight_params = list(var_form.parameters)
+
+        # define interpret functions
+        def interpret_1d(x):
+            return sum((s == "1" for s in f"{x:0b}")) % 2
+
+        self.interpret_1d = interpret_1d
+        self.output_shape_1d = 2  # takes values in {0, 1}
+
+        def interpret_2d(x):
+            return np.array([self.interpret_1d(x), 2 * self.interpret_1d(x)])
+
+        self.interpret_2d = interpret_2d
+        self.output_shape_2d = (
+            2,
+            3,
+        )  # 1st dim. takes values in {0, 1} 2nd dim in {0, 1, 2}
+
+        # define sampler primitives
+        self.sampler = Sampler()
+        self.sampler_shots = Sampler(options={"shots": 100})
```

Comment:
I think adding `seed` to the options when using shots will allow its outcome to be reproducible  which we need for unit tests.